pkg/agent/storage/utils.go

```diff
@@ -132,7 +133,11 @@ func GetProvider(providers map[Protocol]Provider, protocol Protocol) (Provider,
 				}),
 			}
 		}
-
+	case HTTPS:
+		var httpsClient HTTPSClient = &http.Client{}
```

Comment:
Thinking out loud here, but how would you see the agent supporting HTTPS endpoints with self-signed certificates? 

I can see how this may be out of scope of this PR, but would be interesting to see how this could be implemented as a next step.

Reply:
Could you elaborate more on this? 

I'm not too familiar with self-signed certificates. Would the agent behaviour be drastically different?

Reply:
By default, the `net/http` package in Go will reject sending requests to `https://` endpoints signed with self-signed certificates (as it can't verify them). To override that behaviour, you need to change the `http.Transport` used, by one which has the `InsecureSkipVerify` flag enabled in the `TLSClientConfig` object. 

However, this is only useful for self-signed certs. With regular ones, you do want to verify them. I'm wondering whether this could be a field in one of the credentials secrets which gets picked by the agent?

Reply:
I'm wondering whether we should open this as a separate issue?

Reply:
I think if the nodes are installed with root certificates then the https client should just work, otherwise you'd need to load the certificate and set the TLS transport on the client.